frontend/src/services/settings.ts

```diff
@@ -66,36 +86,40 @@ export const getSettings = (): Settings => {
  */
 export const saveSettings = (settings: Partial<Settings>) => {
   Object.keys(settings).forEach((key) => {
-    const isValid = validKeys.includes(key as keyof Settings);
+    const isValidKey = validKeys.includes(key as keyof Settings);
     const value = settings[key as keyof Settings];
-
-    if (isValid && value) localStorage.setItem(key, value);
+    if (isValidKey && typeof value !== "undefined")
+      localStorage.setItem(key, value);
   });
   localStorage.setItem("SETTINGS_VERSION", LATEST_SETTINGS_VERSION.toString());
 };
 
 /**
- * Get the difference between the current settings and the provided settings.
+ * Get the difference between two sets of settings.
  * Useful for notifying the user of exact changes.
  *
  * @example
- * // Assuming the current settings are: { LLM_MODEL: "gpt-3.5", AGENT: "MonologueAgent", LANGUAGE: "en" }
- * const updatedSettings = getSettingsDifference({ LLM_MODEL: "gpt-3.5", AGENT: "OTHER_AGENT", LANGUAGE: "en" });
+ * // Assuming the current settings are:
+ * const updatedSettings = getSettingsDifference(
+ *  { LLM_MODEL: "gpt-3.5", AGENT: "MonologueAgent", LANGUAGE: "en" },
+ *  { LLM_MODEL: "gpt-3.5", AGENT: "OTHER_AGENT", LANGUAGE: "en" }
+ * )
  * // updatedSettings = { AGENT: "OTHER_AGENT" }
  *
- * @param settings - the settings to compare
- * @returns the updated settings
+ * @returns only the settings from `newSettings` that are different from `oldSettings`.
  */
-export const getSettingsDifference = (settings: Partial<Settings>) => {
-  const currentSettings = getSettings();
+export const getSettingsDifference = (
```

Comment:
This change is for testability: The test for SettingsModal mocks getSettings.
Unfortunately, the internal call to getSettings in getSettingsDifferencein could not be mocked, because of [this](https://github.com/vitest-dev/vitest/discussions/3548).

I think the new way is also better in terms of separation of concerns.